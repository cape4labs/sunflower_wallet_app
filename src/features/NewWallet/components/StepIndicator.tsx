import { View } from 'react-native';

interface StepIndicatorProps {
  totalSteps: number;
  currentStep: number;
}

export function StepIndicator({ totalSteps, currentStep }: StepIndicatorProps) {
  return (
    <View className="flex-row justify-center mt-6 space-x-2">
      {Array.from({ length: totalSteps }).map((_, indx) => {
        const isActive = indx + 1 === currentStep;
        return (
          <View
            key={indx}
            className={`w-3 h-3 rounded-full mx-1 ${
              isActive ? 'bg-white' : 'bg-gray-500 opacity-50'
            }`}
          />
        );
      })}
    </View>
  );
}
