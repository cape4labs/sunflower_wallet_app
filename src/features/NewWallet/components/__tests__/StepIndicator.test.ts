import { render } from '@testing-library/react-native';
import { StepIndicator } from '../StepIndicator';

it('StepIndicator snapshot', () => {
  const tree = render(StepIndicator({ totalSteps: 5, currentStep: 3 })).toJSON();
  expect(tree).toMatchSnapshot();
});
