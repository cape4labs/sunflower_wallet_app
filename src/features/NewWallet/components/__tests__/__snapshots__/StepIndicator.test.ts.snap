// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepIndicator snapshot 1`] = `
<View
  className="flex-row justify-center mt-6 space-x-2"
>
  <View
    className="w-3 h-3 rounded-full mx-1 bg-gray-500 opacity-50"
  />
  <View
    className="w-3 h-3 rounded-full mx-1 bg-gray-500 opacity-50"
  />
  <View
    className="w-3 h-3 rounded-full mx-1 bg-white"
  />
  <View
    className="w-3 h-3 rounded-full mx-1 bg-gray-500 opacity-50"
  />
  <View
    className="w-3 h-3 rounded-full mx-1 bg-gray-500 opacity-50"
  />
</View>
`;
